src/language.rs

```diff
@@ -44,11 +58,67 @@ impl SupportedLanguage for SupportedLanguageTypescript {
         tree_sitter_typescript::language_tsx()
     }
 
+    fn name(&self) -> SupportedLanguageName {
+        SupportedLanguageName::Typescript
+    }
+
     fn name_for_ignore_select(&self) -> &'static str {
         "ts"
     }
+
+    fn extensions(&self) -> Vec<&'static str> {
+        vec!["ts", "tsx"]
+    }
 }
 
 pub fn get_typescript_language() -> SupportedLanguageTypescript {
     SupportedLanguageTypescript
 }
+
+pub struct SupportedLanguageJavascript;
+
+impl SupportedLanguage for SupportedLanguageJavascript {
+    fn language(&self) -> Language {
+        tree_sitter_javascript::language()
+    }
+
+    fn name(&self) -> SupportedLanguageName {
+        SupportedLanguageName::Javascript
+    }
+
+    fn name_for_ignore_select(&self) -> &'static str {
+        "js"
+    }
+
+    fn extensions(&self) -> Vec<&'static str> {
+        vec!["js", "jsx", "vue", "cjs", "mjs"]
+    }
+}
+
+pub fn get_javascript_language() -> SupportedLanguageJavascript {
+    SupportedLanguageJavascript
+}
+
+pub fn get_all_supported_languages() -> HashMap<SupportedLanguageName, Box<dyn SupportedLanguage>> {
```

Comment:
I started to think about making this eg a `static` `Lazy` singleton and then decided I was being premature optimization-y